app/scripts/services/spouse.js

```diff
@@ -0,0 +1,35 @@
+'use strict';
+
+angular.module('confRegistrationWebApp')
+  .factory('spouse', function ($http, $cacheFactory) {
+    var spouseRegistrationCache = $cacheFactory('spouseRegistration');
+
+    var spouse = {
+      // Preload a spouse registration so that calling getSpouseRegistration for that conference will complete sooner
+      // This functionality is provided because looking up a spouse registration via getSpouseRegistration is a lengthy operation
+      preloadSpouseRegistration: function (conferenceId) {
+        return spouse.getSpouseRegistration(conferenceId);
```

Comment:
They are identical to hide an implementation detail. Currently, `getSpouseRegistration` uses the built-in `$http` caching mechanism to cache responses. So, caching is already built-in to `getSpouseRegistration`, and `preloadSpouseRegistration` can just mirror `getSpouseRegistration`. But we could conceivably have a different caching mechanism where `preloadSpouseRegistration` and `getSpouseRegistration` would differ. And I don't want clients of the `spouse` service to know the implementation details of `getSpouseRegistration` to know that it caches responses. That is my reasoning behind having two separate functions. What do you think?